local_planner/test/test_local_planner.cpp

```diff
@@ -0,0 +1,216 @@
+#include <gtest/gtest.h>
+
+#include <cmath>
+
+#include "../src/nodes/local_planner.h"
+
+// Stateless tests:
+// Create some hardcoded scan data of obstacles in different positions
+// For each one check that the planner response is correct
+
+class LocalPlannerTests : public ::testing::Test {
+ public:
+  LocalPlanner planner;
+
+  void SetUp() override {
+    ros::Time::init();
+
+    avoidance::LocalPlannerNodeConfig config =
+        avoidance::LocalPlannerNodeConfig::__getDefault__();
+    config.send_obstacles_fcu_ = true;
+    planner.dynamicReconfigureSetParams(config, 1);
+
+    // start with basic pose
+    geometry_msgs::PoseStamped msg;
+    msg.header.seq = 42;
+    msg.header.stamp = ros::Time(500, 0);
+    msg.header.frame_id = 1;
+    msg.pose.position.x = 0;
+    msg.pose.position.y = 0;
+    msg.pose.position.z = 0;
+    msg.pose.orientation.w = 1;
+    planner.currently_armed_ = false;
+    planner.setPose(msg);
+
+    // rise to altitude
+    planner.currently_armed_ = true;
+    msg.pose.position.z = 30;
+    planner.setPose(msg);
+
+    // goal straight in front, 100m away
+    planner.goal_x_param_ = 100;
+    planner.goal_y_param_ = 0;
+    planner.goal_z_param_ = 30;
+    planner.setGoal();
+
+    planner.complete_cloud_.clear();
+  }
+  void TearDown() override {}
+};
+
+TEST_F(LocalPlannerTests, no_obstacles) {
+  // GIVEN: a local planner, a scan with no obstacles, pose and goal
+  planner.complete_cloud_.emplace_back();
+
+  // WHEN: we run the local planner
+  planner.runPlanner();
+  planner.runPlanner();
```

Comment:
why is it necessary to run the planner two times?

Reply:
`runPlanner()` only checks that it is at the correct altitude (`reach_altitude_`) at the end of the `runPlanner()`, so the first time it is run it doesn't actually do anything. I think there is an `else` which can probably be removed in `determineStrategy()` line 183 and instead checking `reach_altitude_` again, but I want a bit more testing before we start changing the logic of the planner.